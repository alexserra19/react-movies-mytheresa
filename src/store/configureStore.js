import AppReducer from './reducers'
import createSagaMiddleware from "redux-saga";
import rootSaga from "./saga";
import { createStore, applyMiddleware } from 'redux';

export default () => {
  const sagaMiddleware = createSagaMiddleware()
  const store = createStore(
    AppReducer,
    applyMiddleware(sagaMiddleware)
  );
  sagaMiddleware.run(rootSaga)

  return store
}