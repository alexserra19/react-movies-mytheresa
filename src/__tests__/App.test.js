import React from 'react';
import { render } from '@testing-library/react';
import App from '../App';
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux';

const middlewares = []
const mockStore = configureStore(middlewares)


const store = mockStore(
  {
    mediaReducer: {
      popularMovies: [],
      topRatedMovies: [],
      upcomingMovies: [],
      selectedMedia: {},
      isLoading: false,
      wishList: []
    }
  }
)


test('renders App Correctly', () => {
  const wrapper = render(
    <Provider store={store}>
      <App />
    </Provider>
  );
  expect(wrapper).toMatchSnapshot();
});
