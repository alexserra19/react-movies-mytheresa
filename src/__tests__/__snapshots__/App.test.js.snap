// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App Correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="ant-layout-header header-container"
      >
        <div
          class="ant-row"
        >
          <div
            class="ant-col web-title ant-col-xs-20 ant-col-md-4"
          >
            <a
              href="/home"
            >
              <h3>
                The Movie Database
              </h3>
            </a>
          </div>
          <div
            class="ant-col ant-col-xs-4 ant-col-md-20"
          >
            <ul
              class="ant-menu-overflow ant-menu ant-menu-root ant-menu-horizontal ant-menu-dark"
              data-menu-list="true"
              role="menu"
              tabindex="0"
            >
              <li
                class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
                data-menu-id="rc-menu-uuid-test-home"
                role="menuitem"
                style="opacity: 1; order: 0;"
                tabindex="-1"
              >
                <span
                  class="ant-menu-title-content"
                >
                  <a
                    href="/home"
                  >
                    Home
                  </a>
                </span>
              </li>
              <li
                class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
                role="menuitem"
                style="opacity: 1; order: 1;"
                tabindex="-1"
              >
                <span
                  class="ant-menu-title-content"
                >
                  <a
                    href="/wishlist"
                  >
                    Wish List
                  </a>
                </span>
              </li>
              <li
                aria-hidden="true"
                class="ant-menu-overflow-item ant-menu-overflow-item-rest ant-menu-submenu ant-menu-submenu-horizontal"
                role="none"
                style="opacity: 0; height: 0px; overflow-y: hidden; order: 9007199254740991; pointer-events: none; position: absolute;"
              >
                <div
                  aria-controls="rc-menu-uuid-test-rc-menu-more-popup"
                  aria-expanded="false"
                  aria-haspopup="true"
                  class="ant-menu-submenu-title"
                  data-menu-id="rc-menu-uuid-test-rc-menu-more"
                  role="menuitem"
                  tabindex="-1"
                >
                  <span
                    aria-label="ellipsis"
                    class="anticon anticon-ellipsis"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="ellipsis"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                      />
                    </svg>
                  </span>
                  <i
                    class="ant-menu-submenu-arrow"
                  />
                </div>
              </li>
            </ul>
            <div
              aria-hidden="true"
              style="display: none;"
            />
          </div>
        </div>
      </header>
      <section
        class="ant-layout layout"
      >
        <main
          class="ant-layout-content layoutContainer"
        >
          <h1>
            Home
          </h1>
          <div>
            <div
              class="ant-row"
            >
              <div
                class="ant-col ant-col-24"
              >
                <div
                  class="ant-row carousel-component"
                >
                  <div
                    class="ant-col ant-col-24"
                  >
                    <h2>
                      Popular Movies
                    </h2>
                  </div>
                  <div
                    class="ant-col ant-col-24 carousel-container"
                  >
                    <div
                      class="ant-carousel"
                    >
                      <div
                        class="slick-slider slick-initialized"
                      >
                        
                        <div
                          class="slick-list"
                        >
                          <div
                            class="slick-track"
                            style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
                          />
                        </div>
                        
                        
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-row"
            >
              <div
                class="ant-col ant-col-24"
              >
                <div
                  class="ant-row carousel-component"
                >
                  <div
                    class="ant-col ant-col-24"
                  >
                    <h2>
                      Top Rated Movies
                    </h2>
                  </div>
                  <div
                    class="ant-col ant-col-24 carousel-container"
                  >
                    <div
                      class="ant-carousel"
                    >
                      <div
                        class="slick-slider slick-initialized"
                      >
                        
                        <div
                          class="slick-list"
                        >
                          <div
                            class="slick-track"
                            style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
                          />
                        </div>
                        
                        
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-row"
            >
              <div
                class="ant-col ant-col-24"
              >
                <div
                  class="ant-row carousel-component"
                >
                  <div
                    class="ant-col ant-col-24"
                  >
                    <h2>
                      Upcoming Movies
                    </h2>
                  </div>
                  <div
                    class="ant-col ant-col-24 carousel-container"
                  >
                    <div
                      class="ant-carousel"
                    >
                      <div
                        class="slick-slider slick-initialized"
                      >
                        
                        <div
                          class="slick-list"
                        >
                          <div
                            class="slick-track"
                            style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
                          />
                        </div>
                        
                        
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </main>
      </section>
      <footer
        class="ant-layout-footer"
        style="text-align: center;"
      >
        The Movie Database MyTheresa Â©2021 Created by Alex Serra
      </footer>
    </div>
  </body>,
  "container": <div>
    <header
      class="ant-layout-header header-container"
    >
      <div
        class="ant-row"
      >
        <div
          class="ant-col web-title ant-col-xs-20 ant-col-md-4"
        >
          <a
            href="/home"
          >
            <h3>
              The Movie Database
            </h3>
          </a>
        </div>
        <div
          class="ant-col ant-col-xs-4 ant-col-md-20"
        >
          <ul
            class="ant-menu-overflow ant-menu ant-menu-root ant-menu-horizontal ant-menu-dark"
            data-menu-list="true"
            role="menu"
            tabindex="0"
          >
            <li
              class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
              data-menu-id="rc-menu-uuid-test-home"
              role="menuitem"
              style="opacity: 1; order: 0;"
              tabindex="-1"
            >
              <span
                class="ant-menu-title-content"
              >
                <a
                  href="/home"
                >
                  Home
                </a>
              </span>
            </li>
            <li
              class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
              role="menuitem"
              style="opacity: 1; order: 1;"
              tabindex="-1"
            >
              <span
                class="ant-menu-title-content"
              >
                <a
                  href="/wishlist"
                >
                  Wish List
                </a>
              </span>
            </li>
            <li
              aria-hidden="true"
              class="ant-menu-overflow-item ant-menu-overflow-item-rest ant-menu-submenu ant-menu-submenu-horizontal"
              role="none"
              style="opacity: 0; height: 0px; overflow-y: hidden; order: 9007199254740991; pointer-events: none; position: absolute;"
            >
              <div
                aria-controls="rc-menu-uuid-test-rc-menu-more-popup"
                aria-expanded="false"
                aria-haspopup="true"
                class="ant-menu-submenu-title"
                data-menu-id="rc-menu-uuid-test-rc-menu-more"
                role="menuitem"
                tabindex="-1"
              >
                <span
                  aria-label="ellipsis"
                  class="anticon anticon-ellipsis"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="ellipsis"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                    />
                  </svg>
                </span>
                <i
                  class="ant-menu-submenu-arrow"
                />
              </div>
            </li>
          </ul>
          <div
            aria-hidden="true"
            style="display: none;"
          />
        </div>
      </div>
    </header>
    <section
      class="ant-layout layout"
    >
      <main
        class="ant-layout-content layoutContainer"
      >
        <h1>
          Home
        </h1>
        <div>
          <div
            class="ant-row"
          >
            <div
              class="ant-col ant-col-24"
            >
              <div
                class="ant-row carousel-component"
              >
                <div
                  class="ant-col ant-col-24"
                >
                  <h2>
                    Popular Movies
                  </h2>
                </div>
                <div
                  class="ant-col ant-col-24 carousel-container"
                >
                  <div
                    class="ant-carousel"
                  >
                    <div
                      class="slick-slider slick-initialized"
                    >
                      
                      <div
                        class="slick-list"
                      >
                        <div
                          class="slick-track"
                          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
                        />
                      </div>
                      
                      
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row"
          >
            <div
              class="ant-col ant-col-24"
            >
              <div
                class="ant-row carousel-component"
              >
                <div
                  class="ant-col ant-col-24"
                >
                  <h2>
                    Top Rated Movies
                  </h2>
                </div>
                <div
                  class="ant-col ant-col-24 carousel-container"
                >
                  <div
                    class="ant-carousel"
                  >
                    <div
                      class="slick-slider slick-initialized"
                    >
                      
                      <div
                        class="slick-list"
                      >
                        <div
                          class="slick-track"
                          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
                        />
                      </div>
                      
                      
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row"
          >
            <div
              class="ant-col ant-col-24"
            >
              <div
                class="ant-row carousel-component"
              >
                <div
                  class="ant-col ant-col-24"
                >
                  <h2>
                    Upcoming Movies
                  </h2>
                </div>
                <div
                  class="ant-col ant-col-24 carousel-container"
                >
                  <div
                    class="ant-carousel"
                  >
                    <div
                      class="slick-slider slick-initialized"
                    >
                      
                      <div
                        class="slick-list"
                      >
                        <div
                          class="slick-track"
                          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
                        />
                      </div>
                      
                      
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </main>
    </section>
    <footer
      class="ant-layout-footer"
      style="text-align: center;"
    >
      The Movie Database MyTheresa Â©2021 Created by Alex Serra
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
